tests/cobbler_collections/system_collection_test.py

```diff
@@ -359,11 +360,371 @@ def test_remove(
 
     # Assert
-    assert name not in system_collection.listing
+    assert system1.uid not in system_collection.listing
     for key, _ in system_collection.indexes.items():
         assert len(system_collection.indexes[key]) == 0
 
 
+@pytest.mark.parametrize(
+    "recursive,expected_exception,expected_result",
+    [
+        (False, pytest.raises(CX), 1),
+        (True, does_not_raise(), 0),
+    ],
+)
+def test_remove_by_image_dependency(
+    cobbler_api: CobblerAPI,
+    create_image: Callable[[], image.Image],
+    create_system: Callable[..., system.System],
+    system_collection: systems.Systems,
+    recursive: bool,
+    expected_exception: Any,
+    expected_result: int,
+):
+    """
+    Validate that a system can be removed from the collection.
+    """
+    # Arrange
+    test_image = create_image()
+    system1 = create_system(image_uid=test_image.uid)
+    system1.interfaces = {  # type: ignore[method-assign]
+        "default": {
+            "ip_address": "192.168.1.1",
+            "ipv6_address": "::1",
+            "dns_name": "example.org",
+            "mac_address": "52:54:00:7d:81:f4",
+        }
+    }
+
+    # Act
+    with expected_exception:
+        cobbler_api.images().remove(test_image, recursive=recursive)
+
+        # Assert
+        assert (system1.uid not in system_collection.listing) == recursive
+        for key, _ in system_collection.indexes.items():
+            assert len(system_collection.indexes[key]) == expected_result
+
+
+@pytest.mark.parametrize(
+    "recursive,expected_exception,expected_result",
+    [
+        (False, pytest.raises(CX), 1),
+        (True, does_not_raise(), 0),
+    ],
+)
+def test_remove_by_menu_image_dependency(
+    cobbler_api: CobblerAPI,
+    create_image: Callable[[], image.Image],
+    create_system: Callable[..., system.System],
+    system_collection: systems.Systems,
+    recursive: bool,
+    expected_exception: Any,
+    expected_result: int,
+):
+    """
+    Validate that a system can be removed from the collection.
+    """
+    # Arrange
+    test_image = create_image()
+    test_menu = cobbler_api.new_menu()
+    test_menu.name = "test_menu"  # type: ignore[method-assign]
+    cobbler_api.menus().add(test_menu)
+    test_image.menu = test_menu.uid  # type: ignore[method-assign]
+    system1 = create_system(image_uid=test_image.uid)
+    system1.interfaces = {  # type: ignore[method-assign]
+        "default": {
+            "ip_address": "192.168.1.1",
+            "ipv6_address": "::1",
+            "dns_name": "example.org",
+            "mac_address": "52:54:00:7d:81:f4",
+        }
+    }
+
+    # Act
+    with expected_exception:
+        cobbler_api.menus().remove(test_menu, recursive=recursive)
+
+        # Assert
+        assert (system1.uid not in system_collection.listing) == recursive
+        for key, _ in system_collection.indexes.items():
+            assert len(system_collection.indexes[key]) == expected_result
```

Comment:
I found more errors in the tests. In this code, when an exception occurs, the checks are not performed at all.